src/Logging.XUnit/XUnitLoggerProvider.cs

```diff
@@ -58,7 +26,20 @@ public XUnitLoggerProvider(ITestOutputHelperAccessor accessor, XUnitLoggerOption
         }
 
         /// <inheritdoc />
-        public virtual ILogger CreateLogger(string categoryName) => new XUnitLogger(categoryName, _accessor, _options);
+        public virtual ILogger CreateLogger(string categoryName)
+        {
+            if (_outputHelperAccessor != null)
+            {
+                return new XUnitLogger(categoryName, _outputHelperAccessor, _options);
+            }
+
+            if (_messageSinkAccessor != null)
+            {
+                return new XUnitLogger(categoryName, _messageSinkAccessor, _options);
+            }
+
+            throw new InvalidOperationException($"Either {nameof(_outputHelperAccessor)} or {nameof(_messageSinkAccessor)} must not be null.");
```

Comment:
Can you add a test case to verify this please?

Reply:
Actually I can't. This branch of the code can not be reached since `XUnitLoggerProvider` is initialized with either a non null `IMessageSinkAccessor` or a non null `ITestOutputHelperAccessor`. This `InvalidOperationException` is there only to make the compiler aware that this code path is unreachable. I have updated the InvalidOperationException message to reflect that in a752efb30e8988ef3c6cc1b12f89e8c12b29b4b7.